conda/common/configuration.py

```diff
@@ -390,6 +390,7 @@ def make_raw_parameters(cls, source, from_map):
         return EMPTY_MAP
 
     @classmethod
+    @cache
```

Comment:
This assumes that the contents of config files do not change during the execution of one conda command. I think this is a safe assumption to make? Though I can see that being wrong. For example, is conda meant to be able to run some actions after a `conda config --file` command updates a given command file?

Using this approach, if conda needs to re-initialize the context due to a config file being updated, then the calling code must clear this cache. For example `YamlRawParameter.make_raw_parameters_from_file.cache_clear()`.